Kernel/Memory/MemoryManager.cpp

```diff
@@ -245,6 +245,28 @@ UNMAP_AFTER_INIT void MemoryManager::parse_memory_map()
     m_global_data.with([&](auto& global_data) {
         global_data.used_memory_ranges.ensure_capacity(4);
 #if ARCH(X86_64)
+        // NOTE: We don't touch the first 1 MiB of RAM on x86-64 even if it's usable as indicated
+        // by a certain memory map. There are 2 reasons for this:
+        //
+        // The first reason is specified for Linux doing the same thing in
+        // https://cateee.net/lkddb/web-lkddb/X86_RESERVE_LOW.html -
+        // "By default we reserve the first 64K of physical RAM, as a number of BIOSes are known
+        //  to corrupt that memory range during events such as suspend/resume or monitor cable insertion,
+        //  so it must not be used by the kernel."
+        //
+        // Linux also allows configuring this knob in compiletime for this reserved range length, that might
+        // also include the EBDA and other potential ranges in the first 1 MiB that could be corrupted by the BIOS:
+        // "You can set this to 4 if you are absolutely sure that you trust the BIOS to get all its memory
+        //  reservations and usages right. If you know your BIOS have problems beyond the default 64K area,
+        //  you can set this to 640 to avoid using the entire low memory range."
+        //
+        // The second reason is that the first 1 MiB memory range should also include the actual BIOS blob
+        // together with possible execution blob code for various option ROMs, which should not be touched
+        // by our kernel.
+        //
+        // **To be completely on the safe side** and never worry about where the EBDA is located, how BIOS might
+        // corrupt the low memory range during power state changing, other bad behavior of some BIOS might change
+        // a value in the very first 64k bytes of RAM, etc - we should just ignore this range completely.
```

Comment:
We usually do this kind of thing for a very broad concept (like how the Graphics subsystem works) - this is a rather niche thing so I rather keep it. If more people think this is the appropriate thing to do than sure, I will write a document.

Reply:
i see it in a kernel documentation, which – for most of its parts – is lacking. thatʼs at least how i feel about it. yet this project is not (by its definition) an educational project… so agreeing! 😅

i feel that copying the above would be enough so that wouldnʼt incur more work (hopefully!). iʼm with you as for the other people needs—yet i hope they will raise their hands or shout out! 😂

Reply:
This niche information is only relevant when you're digging through the memory manager code. If we create a deeper dive " kernel developer's guide", maybe it could fit. But for now it's so specific it doesn't warrant a document.

Reply:
Cool, so I consider this resolved now.